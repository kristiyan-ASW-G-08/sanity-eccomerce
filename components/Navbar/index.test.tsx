// import React from 'react';
// import { render } from '@testing-library/react';
// import Navbar from '.';
// import fileMock from '../../mocks/fileMock';
// import { Provider } from 'react-redux';
// import { RouterContext } from 'next/dist/shared/lib/router-context';
// import createMockRouter from '../../testUtilities/createMockRouter';
// import Store from '@/store/index';
// import { login } from '@/slices/AuthSlice';

// describe('Navbar', () => {
//   afterEach(() => jest.resetAllMocks());
//   afterAll(() => jest.restoreAllMocks());
//   it('render unauthenticated ', async () => {
//     const { container } = render(<Navbar />, {
//       wrapper: ({ children }) => (
//         <Provider store={Store}>
//           <RouterContext.Provider value={createMockRouter({})}>
//             {children}
//           </RouterContext.Provider>
//         </Provider>
//       ),
//     });

//     expect(container).toBeTruthy();
//     expect(container).toMatchSnapshot();
//   });
//   it('render authenticated ', async () => {
//     Store.dispatch(
//       //@ts-ignore
//       login({
//         token: '',
//         //@ts-ignore
//         user: {},
//       }),
//     );
//     const { container } = render(<Navbar />, {
//       wrapper: ({ children }) => (
//         <Provider store={Store}>
//           <RouterContext.Provider value={createMockRouter({})}>
//             {children}
//           </RouterContext.Provider>
//         </Provider>
//       ),
//     });

//     expect(container).toBeTruthy();
//     expect(container).toMatchSnapshot();
//   });
// });
