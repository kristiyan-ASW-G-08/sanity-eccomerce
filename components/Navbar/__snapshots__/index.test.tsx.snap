// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Navbar render authenticated  1`] = `
<div>
  <nav
    class=" mx-auto p-7 w-screen border-b-4 bg-slate-700 border-red-400 flex flex-col justify-center "
  >
    <form
      class="mb-6 flex justify-center align-center "
    >
      <p
        class="text-gray-400  absolute top-7 left-7 py-2 px-4"
      >
        <svg
          aria-hidden="true"
          class="svg-inline--fa fa-magnifying-glass "
          data-icon="magnifying-glass"
          data-prefix="fas"
          focusable="false"
          height="15"
          role="img"
          viewBox="0 0 512 512"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            d="M500.3 443.7l-119.7-119.7c27.22-40.41 40.65-90.9 33.46-144.7C401.8 87.79 326.8 13.32 235.2 1.723C99.01-15.51-15.51 99.01 1.724 235.2c11.6 91.64 86.08 166.7 177.6 178.9c53.8 7.189 104.3-6.236 144.7-33.46l119.7 119.7c15.62 15.62 40.95 15.62 56.57 0C515.9 484.7 515.9 459.3 500.3 443.7zM79.1 208c0-70.58 57.42-128 128-128s128 57.42 128 128c0 70.58-57.42 128-128 128S79.1 278.6 79.1 208z"
            fill="currentColor"
          />
        </svg>
      </p>
      <input
        class="bg-gray-200 appearance-none border-2 border-gray-200 rounded w-full py-1 px-10 text-slate-700  focus:outline-none focus:bg-white focus:border-red-400 "
        name="search"
        placeholder="Search Product Here"
        type="search"
      />
    </form>
    <ul
      class="absolute translate-y-52 w-full bg-neutral-50 z-10"
      data-testid="list"
    />
    <div
      class="flex items-center justify-between "
    >
      <div>
        <span
          style="box-sizing: border-box; display: inline-block; overflow: hidden; background: none; opacity: 1; border: 0px; margin: 0px; padding: 0px; position: relative; max-width: 100%;"
        >
          <span
            style="box-sizing: border-box; display: block; background: none; opacity: 1; border: 0px; margin: 0px; padding: 0px; max-width: 100%;"
          >
            <img
              alt=""
              aria-hidden="true"
              src="data:image/svg+xml,%3csvg%20xmlns=%27http://www.w3.org/2000/svg%27%20version=%271.1%27%20width=%2740%27%20height=%2720%27/%3e"
              style="display: block; max-width: 100%; background: none; opacity: 1; border: 0px; margin: 0px; padding: 0px;"
            />
          </span>
          <img
            alt="PodCasting Logo"
            data-nimg="intrinsic"
            decoding="async"
            src="data:image/gif;base64,R0lGODlhAQABAIAAAAAAAP///yH5BAEAAAAALAAAAAABAAEAAAIBRAA7"
            style="position: absolute; top: 0px; left: 0px; bottom: 0px; right: 0px; box-sizing: border-box; padding: 0px; margin: auto; display: block; width: 0px; height: 0px; min-width: 100%; max-width: 100%; min-height: 100%; max-height: 100%;"
          />
          <noscript />
        </span>
      </div>
      <ul
        class=" md:flex  hidden space-x-6"
      >
        <li
          class=""
        >
          <a
            class="text-neutral-50 hover:text-red-400 font-bold text-red-400 "
            href="/"
          >
            Home
          </a>
        </li>
        <div>
          <button
            class="text-neutral-50 font-bold hover:text-red-400"
          >
            Categories
          </button>
          <ul
            class="bg-slate-700 mt-10 rounded px-5 py-4 absolute z-10 hidden"
          >
            <li
              class=""
            >
              <a
                class="w-full   text-neutral-50 animate-fade hover:text-red-400  "
                href="/products?category=Laptops%20and%20Computers"
              >
                Laptops and Computers
              </a>
            </li>
            <li
              class=""
            >
              <a
                class="w-full text-neutral-50 animate-fade hover:text-red-400  "
                href="/products?category=Peripherals"
              >
                Peripherals
              </a>
            </li>
            <li
              class=""
            >
              <a
                class="w-full  text-neutral-50 animate-fade hover:text-red-400  "
                href="/products?category=Phones%20and%20Tablets"
              >
                Phones and Tablets
              </a>
            </li>
            <li
              class=""
            >
              <a
                class="w-full text-neutral-50 animate-fade hover:text-red-400  "
                href="/products?category=Audio"
              >
                Audio
              </a>
            </li>
            <li
              class=""
            >
              <a
                class="w-full text-neutral-50 animate-fade hover:text-red-400  "
                href="/products?category=TV"
              >
                TV
              </a>
            </li>
          </ul>
        </div>
        <li
          class=""
        >
          <a
            class="text-neutral-50 hover:text-red-400 font-bold  "
            href="/deals"
          >
            Contacts
          </a>
        </li>
        <a
          href="/basket"
        >
          <button>
            <svg
              aria-hidden="true"
              class="svg-inline--fa fa-cart-shopping text-2xl text-neutral-50"
              data-icon="cart-shopping"
              data-prefix="fas"
              focusable="false"
              role="img"
              viewBox="0 0 576 512"
              xmlns="http://www.w3.org/2000/svg"
            >
              <path
                d="M96 0C107.5 0 117.4 8.19 119.6 19.51L121.1 32H541.8C562.1 32 578.3 52.25 572.6 72.66L518.6 264.7C514.7 278.5 502.1 288 487.8 288H170.7L179.9 336H488C501.3 336 512 346.7 512 360C512 373.3 501.3 384 488 384H159.1C148.5 384 138.6 375.8 136.4 364.5L76.14 48H24C10.75 48 0 37.25 0 24C0 10.75 10.75 0 24 0H96zM128 464C128 437.5 149.5 416 176 416C202.5 416 224 437.5 224 464C224 490.5 202.5 512 176 512C149.5 512 128 490.5 128 464zM512 464C512 490.5 490.5 512 464 512C437.5 512 416 490.5 416 464C416 437.5 437.5 416 464 416C490.5 416 512 437.5 512 464z"
                fill="currentColor"
              />
            </svg>
            <p
              class="absolute p-3 h-4 w-4 flex items-center justify-center text-neutral-50 rounded-full bg-red-400 -translate-y-10 translate-x-4"
            >
              0
            </p>
          </button>
        </a>
        <li
          class=""
        >
          <a
            class="bg-neutral-50 text-red-400 px-4 py-1 rounded-full hover:text-red-400 font-bold hover:bg-slate-700  "
            href="/login"
          >
            Login
          </a>
        </li>
        <li
          class=""
        >
          <a
            class="bg-red-400 text-neutral-50 px-4 py-1 rounded-full hover:text-red-400 font-bold hover:bg-slate-700  "
            href="/sign-up"
          >
            Sign Up
          </a>
        </li>
      </ul>
      <button
        class="text-neutral-50 text-lg hover:text-red-400 sm:hidden block"
      >
         
        <svg
          aria-hidden="true"
          class="svg-inline--fa fa-bars "
          data-icon="bars"
          data-prefix="fas"
          focusable="false"
          height="30"
          role="img"
          viewBox="0 0 448 512"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            d="M0 96C0 78.33 14.33 64 32 64H416C433.7 64 448 78.33 448 96C448 113.7 433.7 128 416 128H32C14.33 128 0 113.7 0 96zM0 256C0 238.3 14.33 224 32 224H416C433.7 224 448 238.3 448 256C448 273.7 433.7 288 416 288H32C14.33 288 0 273.7 0 256zM416 448H32C14.33 448 0 433.7 0 416C0 398.3 14.33 384 32 384H416C433.7 384 448 398.3 448 416C448 433.7 433.7 448 416 448z"
            fill="currentColor"
          />
        </svg>
      </button>
    </div>
  </nav>
  <nav
    class="bg-slate-700  px-2  z-10 absolute w-full hidden"
  >
    <ul
      class=" flex flex-col md:hidden  "
    >
      <li
        class="px-5 mb-3"
      >
        <a
          class="text-neutral-50 hover:text-red-400 font-bold text-red-400 "
          href="/"
        >
          Home
        </a>
      </li>
      <li
        class="px-5 mb-3"
      >
        <a
          class="text-neutral-50 hover:text-red-400 font-bold  "
          href="/products"
        >
          Products
        </a>
      </li>
      <li
        class="px-5 mb-3"
      >
        <a
          class="text-neutral-50 hover:text-red-400 font-bold  "
          href="/deals"
        >
          Deals
        </a>
      </li>
      <li
        class="px-5 mb-3"
      >
        <a
          class="text-neutral-50 hover:text-red-400 font-bold  "
          href="/deals"
        >
          About Us
        </a>
      </li>
      <li
        class="px-5 mb-3"
      >
        <a
          class="text-neutral-50 hover:text-red-400 font-bold  "
          href="/deals"
        >
          Contacts
        </a>
      </li>
      <li
        class="px-5 mb-3"
      >
        <a
          class="text-neutral-50 hover:text-red-400 font-bold  "
          href="/login"
        >
          Login
        </a>
      </li>
      <li
        class="px-5 mb-3"
      >
        <a
          class="text-neutral-50 hover:text-red-400 font-bold  "
          href="/sign-up"
        >
          Sign Up
        </a>
      </li>
      <li
        class="px-5 mb-3"
      >
        <a
          class="text-neutral-50 hover:text-red-400 font-bold  "
          href="/create-product"
        >
          New Product
        </a>
      </li>
    </ul>
  </nav>
</div>
`;

exports[`Navbar render unauthenticated  1`] = `
<div>
  <nav
    class=" mx-auto p-7 w-screen border-b-4 bg-slate-700 border-red-400 flex flex-col justify-center "
  >
    <form
      class="mb-6 flex justify-center align-center "
    >
      <p
        class="text-gray-400  absolute top-7 left-7 py-2 px-4"
      >
        <svg
          aria-hidden="true"
          class="svg-inline--fa fa-magnifying-glass "
          data-icon="magnifying-glass"
          data-prefix="fas"
          focusable="false"
          height="15"
          role="img"
          viewBox="0 0 512 512"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            d="M500.3 443.7l-119.7-119.7c27.22-40.41 40.65-90.9 33.46-144.7C401.8 87.79 326.8 13.32 235.2 1.723C99.01-15.51-15.51 99.01 1.724 235.2c11.6 91.64 86.08 166.7 177.6 178.9c53.8 7.189 104.3-6.236 144.7-33.46l119.7 119.7c15.62 15.62 40.95 15.62 56.57 0C515.9 484.7 515.9 459.3 500.3 443.7zM79.1 208c0-70.58 57.42-128 128-128s128 57.42 128 128c0 70.58-57.42 128-128 128S79.1 278.6 79.1 208z"
            fill="currentColor"
          />
        </svg>
      </p>
      <input
        class="bg-gray-200 appearance-none border-2 border-gray-200 rounded w-full py-1 px-10 text-slate-700  focus:outline-none focus:bg-white focus:border-red-400 "
        name="search"
        placeholder="Search Product Here"
        type="search"
      />
    </form>
    <ul
      class="absolute translate-y-52 w-full bg-neutral-50 z-10"
      data-testid="list"
    />
    <div
      class="flex items-center justify-between "
    >
      <div>
        <span
          style="box-sizing: border-box; display: inline-block; overflow: hidden; background: none; opacity: 1; border: 0px; margin: 0px; padding: 0px; position: relative; max-width: 100%;"
        >
          <span
            style="box-sizing: border-box; display: block; background: none; opacity: 1; border: 0px; margin: 0px; padding: 0px; max-width: 100%;"
          >
            <img
              alt=""
              aria-hidden="true"
              src="data:image/svg+xml,%3csvg%20xmlns=%27http://www.w3.org/2000/svg%27%20version=%271.1%27%20width=%2740%27%20height=%2720%27/%3e"
              style="display: block; max-width: 100%; background: none; opacity: 1; border: 0px; margin: 0px; padding: 0px;"
            />
          </span>
          <img
            alt="PodCasting Logo"
            data-nimg="intrinsic"
            decoding="async"
            src="data:image/gif;base64,R0lGODlhAQABAIAAAAAAAP///yH5BAEAAAAALAAAAAABAAEAAAIBRAA7"
            style="position: absolute; top: 0px; left: 0px; bottom: 0px; right: 0px; box-sizing: border-box; padding: 0px; margin: auto; display: block; width: 0px; height: 0px; min-width: 100%; max-width: 100%; min-height: 100%; max-height: 100%;"
          />
          <noscript />
        </span>
      </div>
      <ul
        class=" md:flex  hidden space-x-6"
      >
        <li
          class=""
        >
          <a
            class="text-neutral-50 hover:text-red-400 font-bold text-red-400 "
            href="/"
          >
            Home
          </a>
        </li>
        <div>
          <button
            class="text-neutral-50 font-bold hover:text-red-400"
          >
            Categories
          </button>
          <ul
            class="bg-slate-700 mt-10 rounded px-5 py-4 absolute z-10 hidden"
          >
            <li
              class=""
            >
              <a
                class="w-full   text-neutral-50 animate-fade hover:text-red-400  "
                href="/products?category=Laptops%20and%20Computers"
              >
                Laptops and Computers
              </a>
            </li>
            <li
              class=""
            >
              <a
                class="w-full text-neutral-50 animate-fade hover:text-red-400  "
                href="/products?category=Peripherals"
              >
                Peripherals
              </a>
            </li>
            <li
              class=""
            >
              <a
                class="w-full  text-neutral-50 animate-fade hover:text-red-400  "
                href="/products?category=Phones%20and%20Tablets"
              >
                Phones and Tablets
              </a>
            </li>
            <li
              class=""
            >
              <a
                class="w-full text-neutral-50 animate-fade hover:text-red-400  "
                href="/products?category=Audio"
              >
                Audio
              </a>
            </li>
            <li
              class=""
            >
              <a
                class="w-full text-neutral-50 animate-fade hover:text-red-400  "
                href="/products?category=TV"
              >
                TV
              </a>
            </li>
          </ul>
        </div>
        <li
          class=""
        >
          <a
            class="text-neutral-50 hover:text-red-400 font-bold  "
            href="/deals"
          >
            Contacts
          </a>
        </li>
        <a
          href="/basket"
        >
          <button>
            <svg
              aria-hidden="true"
              class="svg-inline--fa fa-cart-shopping text-2xl text-neutral-50"
              data-icon="cart-shopping"
              data-prefix="fas"
              focusable="false"
              role="img"
              viewBox="0 0 576 512"
              xmlns="http://www.w3.org/2000/svg"
            >
              <path
                d="M96 0C107.5 0 117.4 8.19 119.6 19.51L121.1 32H541.8C562.1 32 578.3 52.25 572.6 72.66L518.6 264.7C514.7 278.5 502.1 288 487.8 288H170.7L179.9 336H488C501.3 336 512 346.7 512 360C512 373.3 501.3 384 488 384H159.1C148.5 384 138.6 375.8 136.4 364.5L76.14 48H24C10.75 48 0 37.25 0 24C0 10.75 10.75 0 24 0H96zM128 464C128 437.5 149.5 416 176 416C202.5 416 224 437.5 224 464C224 490.5 202.5 512 176 512C149.5 512 128 490.5 128 464zM512 464C512 490.5 490.5 512 464 512C437.5 512 416 490.5 416 464C416 437.5 437.5 416 464 416C490.5 416 512 437.5 512 464z"
                fill="currentColor"
              />
            </svg>
            <p
              class="absolute p-3 h-4 w-4 flex items-center justify-center text-neutral-50 rounded-full bg-red-400 -translate-y-10 translate-x-4"
            >
              0
            </p>
          </button>
        </a>
        <li
          class=""
        >
          <a
            class="bg-neutral-50 text-red-400 px-4 py-1 rounded-full hover:text-red-400 font-bold hover:bg-slate-700  "
            href="/login"
          >
            Login
          </a>
        </li>
        <li
          class=""
        >
          <a
            class="bg-red-400 text-neutral-50 px-4 py-1 rounded-full hover:text-red-400 font-bold hover:bg-slate-700  "
            href="/sign-up"
          >
            Sign Up
          </a>
        </li>
      </ul>
      <button
        class="text-neutral-50 text-lg hover:text-red-400 sm:hidden block"
      >
         
        <svg
          aria-hidden="true"
          class="svg-inline--fa fa-bars "
          data-icon="bars"
          data-prefix="fas"
          focusable="false"
          height="30"
          role="img"
          viewBox="0 0 448 512"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            d="M0 96C0 78.33 14.33 64 32 64H416C433.7 64 448 78.33 448 96C448 113.7 433.7 128 416 128H32C14.33 128 0 113.7 0 96zM0 256C0 238.3 14.33 224 32 224H416C433.7 224 448 238.3 448 256C448 273.7 433.7 288 416 288H32C14.33 288 0 273.7 0 256zM416 448H32C14.33 448 0 433.7 0 416C0 398.3 14.33 384 32 384H416C433.7 384 448 398.3 448 416C448 433.7 433.7 448 416 448z"
            fill="currentColor"
          />
        </svg>
      </button>
    </div>
  </nav>
  <nav
    class="bg-slate-700  px-2  z-10 absolute w-full hidden"
  >
    <ul
      class=" flex flex-col md:hidden  "
    >
      <li
        class="px-5 mb-3"
      >
        <a
          class="text-neutral-50 hover:text-red-400 font-bold text-red-400 "
          href="/"
        >
          Home
        </a>
      </li>
      <li
        class="px-5 mb-3"
      >
        <a
          class="text-neutral-50 hover:text-red-400 font-bold  "
          href="/products"
        >
          Products
        </a>
      </li>
      <li
        class="px-5 mb-3"
      >
        <a
          class="text-neutral-50 hover:text-red-400 font-bold  "
          href="/deals"
        >
          Deals
        </a>
      </li>
      <li
        class="px-5 mb-3"
      >
        <a
          class="text-neutral-50 hover:text-red-400 font-bold  "
          href="/deals"
        >
          About Us
        </a>
      </li>
      <li
        class="px-5 mb-3"
      >
        <a
          class="text-neutral-50 hover:text-red-400 font-bold  "
          href="/deals"
        >
          Contacts
        </a>
      </li>
      <li
        class="px-5 mb-3"
      >
        <a
          class="text-neutral-50 hover:text-red-400 font-bold  "
          href="/login"
        >
          Login
        </a>
      </li>
      <li
        class="px-5 mb-3"
      >
        <a
          class="text-neutral-50 hover:text-red-400 font-bold  "
          href="/sign-up"
        >
          Sign Up
        </a>
      </li>
      <li
        class="px-5 mb-3"
      >
        <a
          class="text-neutral-50 hover:text-red-400 font-bold  "
          href="/create-product"
        >
          New Product
        </a>
      </li>
    </ul>
  </nav>
</div>
`;
